src/window.rs

```diff
@@ -438,9 +440,31 @@ pub fn ui_loop() {
                     keycode: received_keycode,
                     ..
                 } => {
+                    if received_keycode.is_some() {
```

Comment:
Also doubling up here?

Reply:
this logic says if there is already a keycode then we can assume that the new keycode is a different key. So we send that keycode to the vector to get processed and then store the new keycode in the temp variable to be paired with a textinput if it becomes available.

Reply:
So in practice no doubling up happens

Reply:
Wait I think there's a typo then. You want to check keycode and not received_keycode right?

Reply:
I'll add some comments and double check

Reply:
yup you're totally right. Doh

Reply:
Fixed and pushed. I also reformatted it slightly for dryness and readability